homework/zadanie4_five_philosophers.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <thread>
+#include <vector>
+#include <chrono>
+#include <mutex>
+#include <string>
+#include <mutex>
+
+using namespace std::chrono_literals;
+
+std::mutex mtx;
+
+class CTable{
+
+public:
+    std::vector<bool> forks_pool;
+
+    CTable(int number_of_forks){
+        for (size_t i=0; i< number_of_forks; i++){
+            forks_pool.push_back(false);
+        }
+    }
+};
+
+class CPhilosoph{
+
+public:
+    std::string name_;
+    std::string output;
+    int number_;
+    CTable table_;
+    int left_fork_;
+    int right_fork_;
+
+    std::chrono::high_resolution_clock::time_point start_of_live_, end_of_live_;
+    size_t time=0;
+
+    CPhilosoph(const CTable& table, std::string name, int number, int left_fork, int right_fork) : table_(table), name_(name), number_(number), left_fork_(left_fork), right_fork_(right_fork){
+        start_of_live_ = std::chrono::high_resolution_clock::now();
+    }
+
+    ~CPhilosoph(){
+        delete this;
+    }
+
+    void eat(){
+        if (table_.forks_pool[left_fork_] == false or table_.forks_pool[right_fork_]){
+            std::lock_guard lg(mtx);
```

Comment:
tu powinniśmy 2 mutexy blokować. Jeśli table będzie zawierać wektor mutexów, to takie coś będzie :)

Reply:
Uzyto std::scoped_lock. Blokuje dwa mutexy jednoczesnie zeby moc zjesc.